File: pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

/**
 * @Author: Liuyu
 * @Date: 2018/12/3 0003 20:34
 * @Version 1.0
 * 键盘敲烂,月薪过万
 */
public interface BrandService {
    List<TbBrand> findAll();

    PageResult findPage(Integer pageNum, Integer pageSize);

    void add(TbBrand tbBrand);

    void update(TbBrand tbBrand);

    TbBrand findOne(Long id);

    void delete(Long[] ids);

    PageResult search(TbBrand brand, Integer pageNum, Integer pageSize);

    List<Map> selectBrandList();
}

File: pinyougou_manager_web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {
    this.findAll = function () {
        return $http.get("../brand/findAll.do")
    }
    this.findPage = function (pageNum,pageSize) {
        return $http.get("../brand/findPage.do?pageNum="+pageNum+"&pageSize="+pageSize)
    }
    this.add = function (entity) {
        return $http.post("../brand/add.do",entity)
    }
    this.update = function (entity) {
        return $http.post("../brand/update.do",entity)
    }
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id="+id)
    }
    this.dele = function (ids) {
        return $http.get("../brand/delete.do?ids="+ids)
    }
    this.search = function (searchEntity, pageNum, pageSize) {
        return $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity)
    }
    this.selectBrandList = function () {
        return $http.get("../brand/selectBrandList.do")
    }
})